packages/tracing/src/transaction.ts

```diff
@@ -1,18 +1,20 @@
 import { getCurrentHub, Hub } from '@sentry/hub';
-import { Event, Measurements, Transaction as TransactionInterface, TransactionContext } from '@sentry/types';
+import {
+  Event,
+  Measurements,
+  Transaction as TransactionInterface,
+  TransactionContext,
+  TransactionMetadata,
+} from '@sentry/types';
 import { dropUndefinedKeys, isInstanceOf, logger } from '@sentry/utils';
 
 import { Span as SpanClass, SpanRecorder } from './span';
 
-interface TransactionMetadata {
-  transactionSampling?: { [key: string]: string | number };
-}
-
 /** JSDoc */
 export class Transaction extends SpanClass implements TransactionInterface {
   public name: string;
 
-  private _metadata: TransactionMetadata = {};
+  public metadata: TransactionMetadata;
```

Comment:
Related to the previous comment. If we already have a setter, do we want to make it public? IMO it seems unnecessary and misleading. I'd either go with the getter/setter or the public attribute.

Reply:
The setter above as you mentioned "merges", so it's helpful method. But I do agree that we could skip it _if_ we'd go with get/set only, and not merge.

Reply:
It feels way simpler to me to just have it public. Honestly, IDK why I made it private to begin with. 

One thing we might think about (in the long list of things we're thinking about for v7) is having a more coherent strategy for what's public and what's private with setters and getters. I don't feel like I have a great intuitive sense for which is better in cases like this.